lib/services/serviceBus/lib/notificationhubservice.js

```diff
@@ -282,16 +282,26 @@ NotificationHubService.prototype.updateRegistration = function (registration, op
 
   var webResource = WebResource.put(this.hubName + '/registrations/' + registration.RegistrationId);
 
-  registration = _.clone(registration);
   var registrationType = registration[Constants.ATOM_METADATA_MARKER]['ContentRootElement'];
   delete registration[Constants.ATOM_METADATA_MARKER];
   delete registration.ExpirationTime;
 
   if (!registration.Expiry) {
-    delete registration.Expiry;
+      delete registration.Expiry;
   }
 
-  var registrationXml = registrationResult.serialize(registrationType, registration);
+  if (registration.RegistrationId != null) {
```

Comment:
In javascript use `!==` please check [this link](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons) for better explanation.

Reply:
Wouldn't you want to check for `undefined` as well? In javascript `null` and `undefined` have different meanings.